src/meta/api/src/schema_api_impl.rs

```diff
@@ -2525,6 +2533,149 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         Ok(CountTablesReply { count })
     }
 
+    async fn list_table_lock_revs(
+        &self,
+        req: ListTableLockRevReq,
+    ) -> Result<Vec<Revision>, KVAppError> {
+        let prefix = format!("{}/{}", TableLockKey::PREFIX, req.table_id);
+        let reply = self.prefix_list_kv(&prefix).await?;
+
+        let mut revisions = vec![];
+        for (k, _) in reply.into_iter() {
+            let lock_key = TableLockKey::from_str_key(&k).map_err(|e| {
+                let inv = InvalidReply::new("list_table_lock_revs", &e);
+                let meta_net_err = MetaNetworkError::InvalidReply(inv);
+                MetaError::NetworkError(meta_net_err)
+            })?;
+
+            revisions.push(lock_key.revision);
+        }
+        Ok(revisions)
+    }
+
+    async fn upsert_table_lock_rev(
+        &self,
+        req: UpsertTableLockRevReq,
+    ) -> Result<UpsertTableLockRevReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let table_id = req.table_id;
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+
+        let revision = loop {
+            trials.next().unwrap()?;
+
+            let tbid = TableId { table_id };
+            let (_tb_meta_seq, tb_meta) = get_table_by_id_or_err(self, &tbid, ctx).await?;
+
+            debug!(
+                ident = display(&tbid),
+                table_meta = debug(&tb_meta),
+                "upsert_table_lock_rev"
+            );
+
+            let revision = if let Some(rev) = req.revision {
+                rev
+            } else {
+                // generate a new revision.
+                fetch_id(self, IdGenerator::table_lock_id()).await?
+            };
+
+            let lock_key = TableLockKey { table_id, revision };
+
+            let (lock_key_seq, _): (_, Option<EmptyProto>) = get_pb_value(self, &lock_key).await?;
+
+            if lock_key_seq > 0 && req.revision.is_none() {
+                // lock key has been created by other process.
+                continue;
+            }
```

Comment:
It is impossible there is a duplicate lock_key, since the created revision is globally unique.

It looks like there should be two methods for creating a lock and for extending a lock lease respectively.

When creating a lock, it just assert the lock seq to be `0`.
When extending the lease, it asserts a specific lock seq.